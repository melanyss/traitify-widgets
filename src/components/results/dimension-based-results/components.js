export DimensionBasedResults from "./index";
export Dimension from "./dimension";
export Dimensions from "./dimensions";
export PersonalityDetails from "./personality-details";
export PersonalityHeading from "./personality-heading";
export PersonalityTrait from "./personality-trait";
export PersonalityTraits from "./personality-traits";
export PersonalityType from "./personality-type";
export PersonalityTypes from "./personality-types";
export Radar from "./radar";
