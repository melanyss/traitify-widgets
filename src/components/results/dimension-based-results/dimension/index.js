import PropTypes from "prop-types";
import {Component} from "react";
import TraitifyPropType from "lib/helpers/prop-type";
import withTraitify from "lib/with-traitify";
import {rgba} from "lib/helpers/color";
import style from "./style";

class Dimension extends Component{
  static propTypes = {
    getOption: PropTypes.func.isRequired,
    index: PropTypes.number.isRequired,
    traitify: TraitifyPropType.isRequired,
    translate: PropTypes.func.isRequired,
    type: PropTypes.shape({
      personality_type: PropTypes.object.isRequired,
      score: PropTypes.number.isRequired
    }).isRequired
  }
  constructor(props){
    super(props);

    this.state = {showContent: props.index === 0};
  }
  trigger = (e)=>{
    e.preventDefault();

    this.props.traitify.ui.trigger("Dimension.showContent", this, this.props.type.personality_type);
    this.setState((state)=>({showContent: !state.showContent}));
  }
  componentDidMount(){
    this.props.traitify.ui.trigger("Dimension.initialized", this);
  }
  componentDidUpdate(){
    this.props.traitify.ui.trigger("Dimension.updated", this);
  }
  description(suffix){
    const type = this.props.type.personality_type;
    const perspective = (this.props.getOption("perspective") || "firstPerson").replace("Person", "");
    let description = type.details.find((detail)=>(detail.title === `${perspective}_person_${suffix}`));
    description = description && description.body;

    if(description){ return description; }
    description = type.details.find((detail)=>(detail.title === `${perspective}_person_${suffix}`));

    return (description && description.body) || type.description;
  }
  render(){
    const type = this.props.type.personality_type;
    const color = `#${type.badge.color_1}`;
    const benefits = [];
    const pitfalls = [];

    type.details.forEach((detail)=>{
      if(detail.title === "Benefits"){ benefits.push(detail.body); }
      if(detail.title === "Pitfalls"){ pitfalls.push(detail.body); }
    });

    return (
      <li className={style.dimension}>
        <div className={style.main} style={{background: rgba(color, 8.5), borderLeft: `5px solid ${color}`}}>
          <div className={style.side}>
            <p className={style.icon}>
              <img src={type.badge.image_medium} alt="" ariahidden="true" />
            </p>
          </div>
          <div className={style.content}>
            <h2 className={style.title}>
              {type.name} <span style={{color}}>|</span> {this.props.type.score} - {type.level}
            </h2>
            <p className={style.description}>{this.description("short_description")}</p>
            <p className={style.triggerButton}>
              <button
                className={style.trigger}
                style={{background: rgba(color, 30)}}
                onClick={this.trigger}
                type="button"
              >
                {this.props.translate(this.state.showContent ? "show_less" : "show_more")}
              </button>
            </p>
          </div>
        </div>
        {this.state.showContent && (
          <div className={style.details}>
            <div className={style.content} style={{background: rgba(color, 30)}}>
              <div className={style.extendedDesc}>
                <h3>{this.props.translate("extended_description")}</h3>
                <p className={style.description}>{this.description("description")}</p>
              </div>
              <div className={style.detail}>
                <h4 className={style.benefits}>{this.props.translate("potential_benefits")}</h4>
                <ul>
                  {benefits.map((benefit)=>(<li key={benefit}>{benefit}</li>))}
                </ul>
              </div>
              <div className={style.detail}>
                <h4 className={style.pitfalls}>{this.props.translate("potential_pitfalls")}</h4>
                <ul>
                  {pitfalls.map((pitfall)=>(<li key={pitfall}>{pitfall}</li>))}
                </ul>
              </div>
            </div>
          </div>
        )}
      </li>
    );
  }
}

export {Dimension as Component};
export default withTraitify(Dimension);
