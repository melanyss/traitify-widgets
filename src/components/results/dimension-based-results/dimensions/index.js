import Component from "components/traitify-component";
import Dimension from "../dimension";
import style from "./style";

export default class Dimensions extends Component{
  componentDidMount(){
    this.traitify.ui.trigger("Dimensions.initialized", this);
    this.followAssessment();
  }
  componentDidUpdate(){
    this.followAssessment();
  }
  render(){
    if(!this.isReady("results")){ return; }

    const options = this.copyOptions();

    return (
      <ul class={style.dimensions}>
        {this.state.assessment.personality_types.map((type, i)=>{
          return <Dimension type={type} index={i} options={options} />;
        })}
      </ul>
    );
  }
}
