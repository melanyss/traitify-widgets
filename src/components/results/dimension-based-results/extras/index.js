import {h, Component} from "preact";
import style from "./style";

export default class Extras extends Component{
  componentDidMount(){
    this.props.triggerCallback("Extras", "initialized", this);
  }
  render(){
    if(!this.props.resultsReady()) return <div />;

    let props = this.props;
    return (
			<div class={style.extras}>
				<div class={style.extra}>
					<div class={style.content}>
						<div class={style.bar} style={`background: #008dc7;`}></div>
						<h4 class={style.title} style={`color: #008dc7;`}>Complements</h4>
						<p class={style.description}>Sit harum culpa repellendus natus placeat cumque Non dignissimos fugiat sit quos pariatur. Rem distinctio laudantium illo fugiat maxime, animi. Laudantium quasi ex iure recusandae animi. Perferendis nemo dolores ipsam. Dolor optio accusantium quis modi pariatur! Accusantium ad cum repellat sit corrupti in consequuntur Consequatur quia vitae deserunt quos esse. Tempore consequatur id veniam eaque dolor odit Nemo dolores aperiam?</p>
					</div>
				</div>
				<div class={style.extra}>
					<div class={style.content}>
						<div class={style.bar} style={`background: #d04e4a;`}></div>
						<h4 class={style.title} style={`color: #d04e4a;`}>Conflicts</h4>
						<p class={style.description}>Sit harum culpa repellendus natus placeat cumque Non dignissimos fugiat sit quos pariatur. Rem distinctio laudantium illo fugiat maxime, animi. Laudantium quasi ex iure recusandae animi. Perferendis nemo dolores ipsam. Dolor optio accusantium quis modi pariatur! Accusantium ad cum repellat sit corrupti in consequuntur Consequatur quia vitae deserunt quos esse. Tempore consequatur id veniam eaque dolor odit Nemo dolores aperiam?</p>
					</div>
				</div>
				<div class={style.extra}>
					<div class={style.content}>
						<div class={style.bar} style={`background: #32be4b;`}></div>
						<h4 class={style.title} style={`color: #32be4b;`}>Best Work Environments</h4>
						<ul class={style.description}>
							<li>Is fast paced</li>
							<li>Values creativity</li>
							<li>Is fast paced</li>
							<li>Values creativity</li>
							<li>Is fast paced</li>
							<li>Values creativity</li>
							<li>Values creativity</li>
							<li>Is fast paced</li>
							<li>Values creativity</li>
						</ul>
					</div>
				</div>
			</div>
    );
  }
}
