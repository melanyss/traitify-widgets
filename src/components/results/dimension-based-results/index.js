import {Component} from "react";
import Radar from "./radar";
import PersonalityTypes from "./personality-types";
import PersonalityHeading from "./personality-heading";
import PersonalityDetails from "./personality-details";
import Dimensions from "./dimensions";
import PersonalityTraits from "./personality-traits";

export default class DimensionBasedResults extends Component {
  render() {
    return (
      <section>
        <Radar {...this.props} />
        <PersonalityTypes {...this.props} />
        <PersonalityHeading {...this.props} />
        <PersonalityDetails {...this.props} />
        <Dimensions {...this.props} />
        <PersonalityTraits {...this.props} />
      </section>
    );
  }
}
