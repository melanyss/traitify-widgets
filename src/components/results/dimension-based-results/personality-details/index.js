import {Component} from "preact";
import withTraitify from "lib/with-traitify";
import style from "./style";

class PersonalityDetails extends Component{
  componentDidMount(){
    this.props.traitify.ui.trigger("PersonalityDetails.initialized", this);
  }
  componentDidUpdate(){
    this.props.traitify.ui.trigger("PersonalityDetails.updated", this);
  }
  render(){
    if(!this.props.isReady("results")){ return; }

    const personality = this.props.assessment.archetype || {};
    const details = personality.details;
    if(!details){ return; }

    let complement = details.find(d=>d.title === "Complement");
    complement = complement && complement.body;
    let conflict = details.find(d=>d.title === "Conflict");
    conflict = conflict && conflict.body;
    const environments = personality.environments || [];

    return (
      <div class={style.details}>
        {complement && (
          <div class={style.detail}>
            <div class={style.content}>
              <div class={style.bar} style="background: #008dc7;" />
              <h4 class={style.title} style="color: #008dc7;">{this.props.translate("complements")}</h4>
              <p class={style.description}>{complement}</p>
            </div>
          </div>
        )}
        {conflict && (
          <div class={style.detail}>
            <div class={style.content}>
              <div class={style.bar} style="background: #d04e4a;" />
              <h4 class={style.title} style="color: #d04e4a;">{this.props.translate("conflicts")}</h4>
              <p class={style.description}>{conflict}</p>
            </div>
          </div>
        )}
        {environments[0] && (
          <div class={style.detail}>
            <div class={style.content}>
              <div class={style.bar} style="background: #32be4b;" />
              <h4 class={style.title} style="color: #32be4b;">{this.props.translate("best_work_environments")}</h4>
              <ul class={style.description}>
                {environments.map(environment=>(
                  <li>{environment.name}</li>
                ))}
              </ul>
            </div>
          </div>
        )}
      </div>
    );
  }
}

export {PersonalityDetails as Component};
export default withTraitify(PersonalityDetails);
