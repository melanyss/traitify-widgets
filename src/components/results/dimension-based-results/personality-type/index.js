import PropTypes from "prop-types";
import {Component} from "react";
import TraitifyPropType from "lib/helpers/prop-type";
import withTraitify from "lib/with-traitify";
import {rgba} from "lib/helpers/color";
import style from "./style";

class Type extends Component{
  static propTypes = {
    type: PropTypes.shape({
      personality_type: PropTypes.object.isRequired,
      score: PropTypes.number.isRequired
    }).isRequired,
    traitify: TraitifyPropType.isRequired
  }
  trigger = (e)=>{
    e.preventDefault();

    this.props.traitify.ui.trigger("PersonalityType.showContent", this, this.props.type.personality_type);
    this.setState((state)=>({showContent: !state.showContent}));
  }
  componentDidMount(){
    this.props.traitify.ui.trigger("PersonalityType.initialized", this);
  }
  componentDidUpdate(){
    this.props.traitify.ui.trigger("PersonalityType.updated", this);
  }
  render(){
    const type = this.props.type.personality_type;
    const color = `#${type.badge.color_1}`;

    return (
      <li className={style.type} style={{borderLeft: `5px solid ${color}`}}>
        <div className={style.main} style={{background: rgba(color, 8.5)}}>
          <div className={style.content}>
            <h2 className={style.title}>
              {type.name}
              <span className={style.score}>{this.props.type.score} - {type.level}</span>
            </h2>
          </div>
        </div>
      </li>
    );
  }
}

export {Type as Component};
export default withTraitify(Type);
