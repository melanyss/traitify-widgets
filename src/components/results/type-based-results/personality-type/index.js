import {Component} from "preact";
import withTraitify from "lib/with-traitify";
import PersonalityBadge from "../personality-badge";
import style from "./style";

class PersonalityType extends Component{
  componentDidMount(){
    this.props.traitify.ui.trigger("PersonalityType.initialized", this);
  }
  render(){
    if(!this.props.isReady("results")){ return; }

    const type = this.props.assessment.personality_types[0].personality_type;

    return (
      <div class={style.type}>
        <PersonalityBadge type={type} {...this.props} />
        <h3 class={style.name}>{type.name}</h3>
        <p class={style.description}>{type.description}</p>
      </div>
    );
  }
}

export {PersonalityType as Component};
export default withTraitify(PersonalityType);
