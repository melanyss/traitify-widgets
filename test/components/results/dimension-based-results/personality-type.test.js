import {Component} from "components/results/dimension-based-results/personality-type";
import ComponentHandler from "support/component-handler";
import assessment from "support/json/assessment/dimension-based.json";

jest.mock("lib/with-traitify", () => ((value) => value));

describe("PersonalityType", () => {
  let props;

  beforeEach(() => {
    props = {
      type: assessment.personality_types[0],
      ui: {
        current: {},
        off: jest.fn().mockName("off"),
        on: jest.fn().mockName("on"),
        trigger: jest.fn().mockName("trigger")
      }
    };
  });

  describe("callbacks", () => {
    it("triggers initialization", () => {
      new ComponentHandler(<Component {...props} />);

      expect(props.ui.trigger.mock.calls[0][0]).toBe("PersonalityType.initialized");
    });

    it("triggers update", () => {
      const component = new ComponentHandler(<Component {...props} />);
      component.updateProps();

      expect(props.ui.trigger.mock.calls[1][0]).toBe("PersonalityType.updated");
    });
  });

  it("renders component", () => {
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });
});
