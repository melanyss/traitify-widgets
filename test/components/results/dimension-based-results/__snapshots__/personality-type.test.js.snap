// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalityType renders component 1`] = `
<li
  style={
    Object {
      "borderLeft": "5px solid #29B770",
    }
  }
>
  <div
    style={
      Object {
        "background": "rgba(41,183,112,0.085)",
      }
    }
  >
    <div>
      <h2>
        Agreeableness
        <span>
          6
           - 
          Medium
        </span>
      </h2>
    </div>
  </div>
</li>
`;
